uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerHost.java

```diff
@@ -0,0 +1,274 @@
+
+        // Look for the device id dialog box
+        final UiObject deviceIdDialog = UiAutomatorUtils.obtainUiObjectWithResourceId(
+                "android:id/message"
+        );
+
+        Assert.assertTrue(deviceIdDialog.exists());
+
+        try {
+            // get the text on the device id dialog box
+            final String[] deviceIdDialogText = deviceIdDialog.getText().split(":");
+            // look for the device id if present
+            if (deviceIdDialogText[0].equalsIgnoreCase("DeviceId")) {
+                return deviceIdDialogText[1];
+            } else {
+                return null;
+            }
+        } catch (UiObjectNotFoundException e) {
+            throw new AssertionError(e);
+        } finally {
+            // dismiss the dialog
+            UiAutomatorUtils.handleButtonClick("android:id/button1");
+        }
+    }
+
+    @Override
+    public void enableBrowserAccess() {
+        launch();
+
+        if (shouldHandleFirstRun) {
+            handleFirstRun();
+        }
+
+        // Click enable browser access
+        UiAutomatorUtils.handleButtonClick(
+                "com.microsoft.identity.testuserapp:id/buttonInstallCert"
+        );
+
+        final UiDevice device =
+                UiDevice.getInstance(InstrumentationRegistry.getInstrumentation());
+
+        // Install cert
+        final UiObject certInstaller = device.findObject(new UiSelector().packageName("com.android.certinstaller"));
+        certInstaller.waitForExists(FIND_UI_ELEMENT_TIMEOUT);
+        Assert.assertTrue(
+                "Microsoft Authenticator - cert installer dialog appears.",
+                certInstaller.exists()
+        );
+
+        UiAutomatorUtils.handleButtonClick("android:id/button1");
+    }
+
+    @Override
+    public void createPowerLiftIncident() {
+        throw new UnsupportedOperationException("Not supported on Broker Host App :(");
+    }
+
+    @Override
+    public DeviceAdmin getAdminName() {
+        return DeviceAdmin.BROKER_HOST;
+    }
+
+    @Override
+    public void handleFirstRun() {
+        // nothing needed here
+    }
+
+    @Nullable
+    public String getAccountUpn() {
+        launch(); // launch Broker Host app
+
+        if (shouldHandleFirstRun) {
+            handleFirstRun(); // handle first run experience
+        }
+
+        UiAutomatorUtils.handleButtonClick("com.microsoft.identity.testuserapp:id/buttonGetUpn");
+
+        // Look for the UPN dialog box
+        final UiObject showUpnDialog = UiAutomatorUtils.obtainUiObjectWithResourceId(
+                "android:id/message"
+        );
+
+        Assert.assertTrue(showUpnDialog.exists());
+
+        try {
+            // Obtain the text on the UPN dialog box
+            final String[] upnDialogTextParts = showUpnDialog.getText().split(":");
+
+            // get the UPN if it is there, else return null (in case of error)
+            if (upnDialogTextParts[0].equalsIgnoreCase("UPN")) {
```

Comment:
"UPN".equals...

Reply:
Addressed here: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1030/commits/6012894adad9fc883be444247ae7f9d6db733871